src/popover.ts

```diff
@@ -273,6 +273,22 @@ export function apply() {
     return popoverTargetElement;
   };
 
+  function dismissAutoPopovers(
+    event: Event,
+    popoverTargetElement: HTMLElement | null = null,
+  ) {
+    const target = event.target;
+    for (const popover of popovers) {
+      if (
+        popover.matches('[popover="" i].\\:open, [popover=auto i].\\:open') &&
+        popover !== popoverTargetElement &&
+        !queryAncestorAll(target as Element, '[popover]').includes(popover)
```

Comment:
In the case of Popover 3, where the internal link receives auto-focus, is it intentional that `Esc` does not dismiss the popover when focus is on the link? It only dismisses once focus is first moved to another element, not within the popover.

I haven't dug into the spec to see if this is to be expected -- I just found it a bit surprising while testing this out.

Reply:
This sounds like an issue for sure! I didn't notice this but I'll try to reproduce!

Reply:
I managed to reproduce but also discovered another bug, which is fixed in #83 - if we could merge that one first then I can rebase this one on top to bring in some of the refactored code here.